tools/build/pack_windows.py

```diff
@@ -0,0 +1,838 @@
+    LOG.info("+Preparing build root folder ...")
+    global BUILD_ROOT  # The build root directory, like as "/tmp/randomName"
+    # The source code in the tmp directory, like as
+    # "/tmp/randomName/crosswalk-test"
+    global BUILD_ROOT_SRC
+    # The source of the zip operate for all package, like as
+    # "/tmp/randomName/pkg"
+    global BUILD_ROOT_SRC_PKG
+    # The source of the app_package operate for all package, like as
+    # "/tmp/randomName/pkg-app"
+    global BUILD_ROOT_SRC_PKG_APP
+    # The source of the sub app_package operate for all package, like as
+    # "/tmp/randomName/sub-app"
+    global BUILD_ROOT_SRC_SUB_APP
+    global BUILD_ROOT_PKG  # BUILD_ROOT_SRC_PKG + "opt" + PKG_NAME
+    global BUILD_ROOT_PKG_APP  # BUILD_ROOT_SRC_PKG_APP + "opt" + PKG_NAME
+
+    while True:
+        BUILD_ROOT = os.path.join("/tmp", getRandomStr())
+        if os.path.exists(BUILD_ROOT):
+            continue
+        else:
+            break
+
+    BUILD_ROOT_SRC = os.path.join(BUILD_ROOT, PKG_NAME)
+    BUILD_ROOT_SRC_PKG = os.path.join(BUILD_ROOT, "pkg")
+    BUILD_ROOT_SRC_PKG_APP = os.path.join(BUILD_ROOT, "pkg-app")
+    BUILD_ROOT_SRC_SUB_APP = os.path.join(BUILD_ROOT, "sub-app")
+    BUILD_ROOT_PKG = os.path.join(BUILD_ROOT, "pkg", "opt", PKG_NAME)
+    BUILD_ROOT_PKG_APP = os.path.join(BUILD_ROOT, "pkg-app", "opt", PKG_NAME)
+
+    if not doCopy(BUILD_PARAMETERS.srcdir, BUILD_ROOT_SRC):
+        return False
+    else:
+        replaceUserString(
+            BUILD_ROOT_SRC,
+            '*',
+            'TESTER-HOME-DIR',
+            "/home/%s" %
+            BUILD_PARAMETERS.user)
+
+    if not doRemove(
+            glob.glob(os.path.join(BUILD_ROOT_SRC, "%s*.zip" % PKG_NAME))):
+        return False
+
+    return True
+
+
+def doCMD(cmd, time_out=DEFAULT_CMD_TIMEOUT, no_check_return=False):
+    LOG.info("Doing CMD: [ %s ]" % cmd)
+    pre_time = time.time()
+    cmd_proc = subprocess.Popen(args=cmd, shell=True)
+    while True:
+        cmd_exit_code = cmd_proc.poll()
+        elapsed_time = time.time() - pre_time
+        if cmd_exit_code is None:
+            if elapsed_time >= time_out:
+                killProcesses(ppid=cmd_proc.pid)
+                LOG.error("Timeout to exe CMD")
+                return False
+        else:
+            if not no_check_return and cmd_exit_code != 0:
+                LOG.error("Fail to exe CMD")
+                return False
+            break
+        time.sleep(2)
+    return True
+
+def packMsi(build_json=None, app_src=None, app_dest=None, app_name=None):
+    orig_dir = os.getcwd()
+    if not os.path.exists(
+            os.path.join(BUILD_ROOT, "crosswalk-app-tools")):
+        if not doCopy(os.path.join(BUILD_PARAMETERS.pkgpacktools, "crosswalk-app-tools"),
+                      os.path.join(BUILD_ROOT, "crosswalk-app-tools")):
+            return False
+        os.chdir(os.path.join(BUILD_ROOT, "crosswalk-app-tools"))
+        install_cmd = "npm install"
```

Comment:
"npm install" is better to be a precondition. Since you must remove existing node_modules folder before execute "npm install" with root user.